lib/src/main/scala/spinal/lib/com/jtag/sim/JtagVpi.scala

```diff
@@ -0,0 +1,158 @@
+package solsoc
+
+import spinal.core.sim._
+import spinal.lib.com.jtag.Jtag
+
+import java.nio.{ByteBuffer, ByteOrder}
+import spinal.lib.com.jtag.sim.JtagDriver
+
+import java.io.{DataInputStream, DataOutputStream}
+import java.net.ServerSocket
+
+/*
+    * This is a JTAG VPI interface for openocd
+    * It is used to connect openocd to a JTAG interface over TCP
+    * the reference interface implementation is in https://github.com/openocd-org/openocd/blob/master/src/jtag/drivers/jtag_vpi.c
+    * Example openocd config:
+
+source [find interface/jtag_vpi.cfg]
+debug_level 1
+set _CHIPNAME SOC
+jtag newtap $_CHIPNAME cpu -irlen 5 -expected-id 0x10002fff
+# init the CPU
+target create $_CHIPNAME.cpu riscv -chain-position $_CHIPNAME.cpu
+init
+ */
+
+object JtagVpi {
+  def apply(jtag: Jtag, port: Int = 5555, jtagClkPeriod: Long) = fork {
+    val driver = JtagDriver(jtag, jtagClkPeriod)
+    var inputStream: DataInputStream = null
+    var outputStream: DataOutputStream = null
+    class SocketThread extends Thread {
+      val socket = new ServerSocket(port)
+
+      override def run(): Unit = {
+        println("VPI Server started, waiting for connection...")
+        while (true) {
```

Comment:
the code can only handle a single client at the moment (which makes sense, we only need one debugger...), I think it might be wort it to only accept a single connection at one time? 
Looking at the current code this might be as easy as initially blocking on on the socket?

Reply:
The issue with blocking on the socket si that we need some way to sleep, otherwise the simulation cannot go forward without the connection (which is totally fair, you might not want to connect a debugger for this run of the sim). this way with the check of inputStream != null, we sleep until a connection is made. 

right now, the server will drop the first connection once a new one is made.

Reply:
you are right, I overlooked that `fork`d tasks are cooperative... that's why I ended up using a nonblocking accept.

It's not really threadsafe if the connection is dropped currently and a new one is made (since e.g. the outputStream might change while processing the commands sent from the old connection) - not sure how much of an issue this is in reality though...

Reply:
I think I fixed it from the last commit, it was a bit tricky and ugly, but now the server can accept only one connection at a time, but will accept a new connection after the previous has been dropped.